mirascope/tools/system/_computer_use.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import io
+import tarfile
+from abc import ABC
+from contextlib import suppress
+from typing import ClassVar
+
+import docker
+from docker.models.containers import Container
+from pydantic import Field
+
+from mirascope.core import toolkit_tool
+from mirascope.tools.base import (
+    ConfigurableTool,
+    ConfigurableToolKit,
+    _ToolConfig,
+    _ToolSchemaT,
+)
+
+
+class ComputerUseToolkitConfig(_ToolConfig):
+    """Configuration for computer use toolkit"""
+
+    docker_image: str = Field(
+        default="python:3.13-slim", description="Docker image for code execution"
+    )
+    max_memory: str = Field(default="512m", description="Maximum memory allocation")
+    allow_network: bool = Field(default=True, description="Allow network access")
+
+
+class DockerOperation(ConfigurableTool[ComputerUseToolkitConfig, _ToolSchemaT], ABC):
+    """Base class for Docker operations."""
+
+    __config__ = ComputerUseToolkitConfig()
+    __container__: Container | None = None
+
+    @property
+    def _container(self) -> Container:
+        """Returns the Docker container for code execution."""
+        if self.__container__ is None:
+            self.__container__ = self._setup_container()
+        return self.__container__
+
+    def __del__(self) -> None:
```

Comment:
this is going to stop the docker container when this tool gets deleted, but that's not what we want. We want the container to persist until the LLM is done using it so that multiple calls can use the same docker instance.

We probably want the user to have a little more control over the container, right? Maybe something like a `DockerContainer` class that is required as a field of the toolkit? That way the user can create the toolkit wherever and supply it with the same docker container, and then when they are done and the container goes out of scope we'll shut it down. Thoughts?